### PR TITLE
Added problem group entity and methods for migrating the problems

### PR DESCRIPTION
**Pull request by gogo4ds**
Implement ProblemGroup entity and migrator for the problems
Closes https://github.com/SoftUni-Internal/suls-issues/issues/3778

### CODE REVIEW

File: Open Judge System/Data/OJS.Data.Models/Problem.cs [hunk 1]
```diff
@@ -19,6 +19,10 @@ public class Problem : DeletableEntity, IOrderable
 
         public virtual Contest Contest { get; set; }
```
Comment (KristianMariyanov) [inline Open Judge System/Data/OJS.Data.Models/Problem.cs]: Do we have a case. that one problem will have both Problem group and Contest?

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs [hunk 2]
```diff
@@ -139,5 +141,79 @@ public ActionResult RetestCompileTimeoutSubmissionsFromCSharpDbAdvancedExam10Dec
...
+                .ToList();
+
+            var output = new StringBuilder();
+            output.Append($"Done! contests affected: {contests.Count} <ol>");
+
+            foreach (var contest in contests)
+            {
+                foreach (var problem in contest.Problems)
+                {
+                    var problemGroup = new ProblemGroup
```
Comment (KristianMariyanov) [inline Open Judge System/Web/OJS.Web/Controllers/TempController.cs]: What's the point of creating problem group for each problem?

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs [hunk 3]
```diff
@@ -139,5 +141,79 @@ public ActionResult RetestCompileTimeoutSubmissionsFromCSharpDbAdvancedExam10Dec
...
+                .Where(c => c.NumberOfProblemGroups > 1 && !c.ProblemGroups.Any())
+                .ToList();
+
+            var output = new StringBuilder();
+            output.Append($"Done! contests affected: {contests.Count} <ol>");
+
+            foreach (var contest in contests)
+            {
+                output.Append($"<li>Created Problem Groups for Contest: {contest.Name} (ID:{contest.Id})<ol>");
+                for (var i = 0; i < contest.NumberOfProblemGroups; i++)
```
Comment (KristianMariyanov) [inline Open Judge System/Web/OJS.Web/Controllers/TempController.cs]: `NumberOfProblemGroups`  looks like redundant after introducing ProblemGroups, right?

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs (hunk 2 above)
Comment (gogo4ds) [reply Open Judge System/Web/OJS.Web/Controllers/TempController.cs]: To be semantically right as we use Problem Group for grouping problems that can be interchangeable in the Online contest and we want to move towards this logic in every contest. But in the current contests every problem is created as unique and cannot be changed by another, that's why we want them to be in a separate groups.

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs (hunk 3 above)
Comment (gogo4ds) [reply Open Judge System/Web/OJS.Web/Controllers/TempController.cs]: For this issue we are keeping this property, will be refactored further more as part of the bigger issue for this problem https://github.com/SoftUni-Internal/suls-issues/issues/3777

File: Open Judge System/Data/OJS.Data.Models/Problem.cs (hunk 1 above)
Comment (gogo4ds) [reply Open Judge System/Data/OJS.Data.Models/Problem.cs]: For this issue we are keeping this connection, will be refactored further more as part of the bigger issue for this problem https://github.com/SoftUni-Internal/suls-issues/issues/3777

File: Open Judge System/Web/OJS.Web/Controllers/TempController.cs (hunk 3 above)
Comment (KristianMariyanov) [reply Open Judge System/Web/OJS.Web/Controllers/TempController.cs]: :shipit:

**Pull request merged**